pkg/migrations/sql/preinstall/006-set_up_ha.sql

```diff
@@ -0,0 +1,146 @@
+CREATE TABLE SCHEMA_CATALOG.ha_locks
+(
+    cluster_name TEXT PRIMARY KEY,
+    leader       TEXT,
+    lease_start  TIMESTAMPTZ,
+    lease_until  TIMESTAMPTZ
+);
+
+CREATE TABLE SCHEMA_CATALOG.ha_locks_log
+(
+    cluster_name TEXT        NOT NULL,
+    leader       TEXT        NOT NULL,
+    lease_start  TIMESTAMPTZ NOT NULL,
+    lease_until  TIMESTAMPTZ NOT NULL,
+    PRIMARY KEY (cluster_name, lease_start),
+    UNIQUE (cluster_name, leader, lease_start)
+);
+
+
+-- function that trigger to automatically keep the log calls
+CREATE OR REPLACE FUNCTION SCHEMA_CATALOG.ha_locks_audit_fn()
+    RETURNS TRIGGER
+AS
+$func$
+BEGIN
+    INSERT INTO SCHEMA_CATALOG.ha_locks_log (cluster_name, leader, lease_start, lease_until)
+    VALUES (NEW.cluster_name, NEW.leader, NEW.lease_start, NEW.lease_until)
+           -- if lease is extended, but leader didn't change
+    ON CONFLICT (cluster_name, leader, lease_start) DO UPDATE SET lease_until=EXCLUDED.lease_until;
+    --no ON CONFLICT (cluster_name, leader) clause as there should never be conflicts
+    -- on the PK and if there are it's an error
+    RETURN NEW;
+END;
+$func$ LANGUAGE plpgsql VOLATILE;
+
+-- trigger to automatically keep the log
+CREATE TRIGGER ha_locks_audit
+    AFTER INSERT OR UPDATE
+    ON SCHEMA_CATALOG.ha_locks
+    FOR EACH ROW
+EXECUTE PROCEDURE SCHEMA_CATALOG.ha_locks_audit_fn();
+
+-- default values for lease
+INSERT INTO SCHEMA_CATALOG.default(key, value)
+VALUES ('ha_lease_timeout', '1m'),
+       ('ha_lease_refresh', '10s')
+ON CONFLICT (key) DO UPDATE SET value = EXCLUDED.value;
+
+-- ha api functions
+CREATE OR REPLACE FUNCTION SCHEMA_CATALOG.check_insert(cluster TEXT, writer TEXT, min_time TIMESTAMPTZ,
```

Comment:
we should give a better name than check_insert. I am not sure to what as I have a significant amount of code to read.